packages/wonder-blocks-dropdown/src/components/single-select.tsx

```diff
@@ -460,23 +476,20 @@ const SingleSelect = (props: Props) => {
         const selectedItem = items.find(
             (option) => option.props.value === selectedValue,
         );
-        // If nothing is selected, or if the selectedValue doesn't match any
-        // item in the menu, use the placeholder.
 
         let menuContent;
-
         if (selectedItem) {
-            const menuOpenerLabel = getSelectOpenerLabel(
+            const menuStringOrNode = getSelectOpenerLabel(
                 showOpenerLabelAsText,
                 selectedItem.props,
             );
-            // For Custom Option Items with Node Labels, we have to extract
-            // strings to announce
-            const [label, node] = maybeExtractStringFromNode(menuOpenerLabel);
+            // We only need the guaranteed node for SingleSelect here
+            // As the string label for the Announcer is in a useEffect above
+            const [, node] = maybeExtractStringFromNode(menuStringOrNode);
```

Comment:
Since we don't need the `label` from `maybeExtractStringFromNode` here anymore, could we go back to using the return value from `getSelectOpenerLabel` for the `menuContent`?

Reply:
I'll have to play with the types more, or create a separate helper function from `getSelectOpenerLabel`. `SingleSelect` and `MultiSelect` use the same helper functions but this change would require a different type signature. If I use that value from `getSelectOpenerLabel` for `menuContent` in `SingleSelect` as written, it has a type failure:
```
Type 'OpenerStringOrNode' is not assignable to type 'OptionLabel'.
      Type '{ [key: string]: string | Element; }' is not assignable to type 'OptionLabel'.
```

So I'll probably opt for a separate helper function to limit the complexity. I'm not entirely sure why MultiSelect is so much more complicated in this precise way.

Reply:
I got it simplified quite a bit and pushed an update!